### PR TITLE
Remove service name and version requirement for error reporting

### LINKED ISSUE #3263: Allow defaulting for service name and version in diagnostics
One common mistake when configuring diagnostics is to not specify a service name and service version. Error reporting requires these to be non-null. We try to detect them from the platform, but they're only provided by GAE.
We should default to a reasonable value if neither the user nor the platform provides them.

### PR DESCRIPTION
**Pull request by jskeet**
It's fine just not to provide them when logging: the error reporting UI works with no problems, etc.

Fixes #3263.

### CODE REVIEW

**Comment by jskeet** [comment]
(I know this is an area you're not terribly familiar with, Chris - fortunately it's just the same "stop checking" again and again...)

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ProjectTest.cs [hunk 1]
```diff
@@ -86,6 +86,38 @@ public void GetAndCheckServiceName()
             Assert.Equal("No Google App Engine service name was passed in or detected.", ex.Message);
         }
 
+        [Fact]
+        public void GetServiceName()
+        {
+            var serviceName = "my-app";
+            var resourceServiceName = "my-app";
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ProjectTest.cs]: These should not be the same string ;)

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ProjectTest.cs [hunk 2]
```diff
@@ -119,6 +151,38 @@ public void GetAndCheckServiceVersion()
             Assert.Equal("No Google App Engine service version was passed in or detected.", ex.Message);
         }
 
+        [Fact]
+        public void GetServiceVersion()
+        {
+            var serviceVersion = "1.0.0";
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ProjectTest.cs]: Neither should these.

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ProjectTest.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ProjectTest.cs]: This was c-n-p, so I'm sure it's the same everywhere - will fix them all.

**Pull request merged**